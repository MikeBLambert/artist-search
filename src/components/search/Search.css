#searchBar {
  margin: 5em;
  text-align: center;
  box-shadow: 2px 2px grey;
  padding: 35px;
  border: 5px solid black;
  background-color: lightgrey;
  border-radius: 15px;
  label {
    margin-right: 10vw;
    font-family: 'Black Ops One', cursive;
    color: black;
    font-size: 25px;
  }

  input {
    border: 2px solid #666666;
    border-radius: 4px;
    width: 25vw;
  }
}