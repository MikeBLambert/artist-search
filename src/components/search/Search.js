import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Search = ({ updateSearchQuery }) => {
  return (
    <Fragment>
      <label htmlFor='search'>Search for an Artist</label>
      <input name='searchQuery' id='search' onChange={updateSearchQuery} />
    </Fragment>
  );
};

// Search.propTypes = {
//   updateSearchQuery: PropTypes.func.isRequired
// };

export default Search;
