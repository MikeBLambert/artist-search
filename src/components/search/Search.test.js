import React from 'react';
import { shallow } from 'enzyme';
import Search from './Search';

describe('search component', () => {

  it('renders a search component', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has a label and input', () => {
    const wrapper = shallow(<Search />)
    expect(wrapper.find('input').at(0)).toBeTruthy();
    expect(wrapper.find('label').at(0)).toBeTruthy();
  });
});
