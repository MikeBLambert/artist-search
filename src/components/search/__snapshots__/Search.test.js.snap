// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component renders a search component 1`] = `
<Fragment>
  <label
    htmlFor="search"
  >
    Search for an Artist
  </label>
  <input
    id="search"
    name="searchQuery"
  />
</Fragment>
`;
